inspirehep/modules/multieditor/actions.py

```diff
@@ -0,0 +1,376 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2014-2017 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+from __future__ import absolute_import, print_function, division
+
+import re
+
+
+from jsonschema import ValidationError
+from inspire_schemas.api import validate
+
+from inspirehep.utils.record import inspire_diff
+
+
+class Action(object):
+    def __init__(self, keypath, value=None, match_type=None, update_value=None,
+                 conditions=None):
+        """
+        :param keypath: path to value
+        :type keypath: array(string)
+
+       :param value: value to add or update to
+       :type value: string
+
+       :param match_type: type of matching
+       :type match_type: string
+
+       :param update_value: value to replace or delete
+       :type update_value: string
+
+       :param conditions: conditions of the action
+       :type conditions: array(object)
+        """
+        self.keypath = keypath
+        self.value = value
+        self.match_type = match_type
+        self.update_value = update_value
+        self.conditions = conditions
+        self.changed = False
+
+    def get_next_key_and_condition_check(self, record, schema, position, conditions_passed):
+        """
+
+        :param record: record to be processed
+        :param position: recursion level
+        :param conditions_passed: number of conditions that got checked and passed
+        :return: returns the new number of conditions passed,
+         the new key for this recursion and if a condition got checked and the failed the check
+        """
+        key = self.keypath[position]
+        condition_failed = False
+        if self.conditions:
+            for condition in self.conditions:
+                should_check = False
+                if position < len(condition.get('keypath')):
```

Comment:
This used the conditions as received from the frontend.
Now im sanitizing them as you can see in serializers.py.
So instead of .get im just going to access simply the key of the dict.

Reply:
👍